addons/desec.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+true
+SCRIPT_NAME="deSEC Registration"
+SCRIPT_EXPLAINER="This script will automatically register a domain of your liking, and set it to automatically update your external IP address."
+# shellcheck source=lib.sh
+source /var/scripts/fetch_lib.sh || source <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+# Maybe move to lib?
+DEDYNPASS=$(gen_passwd "$SHUF" "a-zA-Z0-9")
+
+# Enter the subdomain
+msg_box "Please enter the subdomain (*example*.dedyn.io) that you want to use"
+while :
+do
+    SUBDEDYN=$(input_box_flow "Please enter the subdomain (*example*.dedyn.io) that you want to use \
+The only allowed characters for the username are:
+'a-z', 'A-Z', and '0-9'")
+    if [[ "$SUBDEDYN" == *" "* ]]
+    then
+        msg_box "Please don't use spaces."
+    elif [ "${SUBDEDYN//[A-Za-z0-9]}" ]
+    then
+        msg_box "Allowed characters for the username are:\na-z', 'A-Z', and '0-9'\n\nPlease try again."
+    else
+        DEDYNDOMAIN="$SUBDEDYN.dedyn.io"
+        break
+    fi
+done
+
+# Email address
+msg_box "You will now be prompted to enter an email address. It's very important that the email address you enter it a 100% valid one.
+
+Every 6 months you will get an email asking you to confrim your domain. If deSeC doesn't get an answer within 3 weeks, you domain will be destroyed!"
+
+VALIDEMAIL=$(input_box_flow "Please enter a valid email address. NOT a fake or a temporary one.")
+
+msg_box "Your account for deSEC password is:
+
+$DEDYNPASS
+
+Please write this down now."
+
+# Register the domain
+curl -X POST https://desec.io/api/v1/auth/ \
+    --header "Content-Type: application/json" --data @- <<EOF
+    {
+      "email": "$VALIDEMAIL",
+      "password": "$DEDYNPASS",
```

Comment:
This is the account password, not the update token. Most users don't need to log into their account, so I recommend disabling password login with `"password": null`.

What you need is an API token for updating the IP address of the domain. That "update token" is displayed after the user has confirmed their email address. You'll have to ask the user to check their mail, confirm their account, and then copy the token from the confirmation page.